tests/setup-snapshot.yml

```diff
@@ -0,0 +1,15 @@
+- hosts: all
+  tasks:
+    - name: Set platform/version specific variables
+      include_role:
+        name: linux-system-roles.logging
+        public: true
```

Comment:
don't you need to use `tasks_from:` here?

Reply:
That's what I've been struggling with... 
In the second commit: 03b00f557cc525e54eebb3af3bc6b3d7f210009e, I had tasks_from like this: https://github.com/linux-system-roles/logging/pull/274/commits/03b00f557cc525e54eebb3af3bc6b3d7f210009e#diff-7eb2a312068fa9ec1be5e435acb86b2def3a99f30b6ff3092f42964283bd9fa8.

The top level task has to have some way to point another task in rsyslog, it contained the "relative" path to another task in rsyslog task like this: https://github.com/linux-system-roles/logging/pull/274/commits/03b00f557cc525e54eebb3af3bc6b3d7f210009e#diff-fde1cadb053b9da84a369cc078950e138d3ccbdb9d3c2f6a6aef1e482fad2f92

It obviously breaks the collection converted format... :(  So, I gave it up and did include_role with no options just to get the ansible facts and the package names.

Reply:
hmm - Then I guess add some sort of logic to logging tasks/main.yml so that when `__snapshot_gather_vars` is `true` the role does nothing except set the facts

Reply:
If the performance is the issue, I compared the duration time between the 2nd commit version https://github.com/linux-system-roles/logging/commit/03b00f557cc525e54eebb3af3bc6b3d7f210009e#diff-7eb2a312068fa9ec1be5e435acb86b2def3a99f30b6ff3092f42964283bd9fa8 and this one. There were almost no difference... But probably, we need some way to avoid the risk to bring in an unexpected bug... Let me think about it...

Reply:
> If the performance is the issue, I compared the duration time between the 2nd commit version [03b00f5#diff-7eb2a312068fa9ec1be5e435acb86b2def3a99f30b6ff3092f42964283bd9fa8](https://github.com/linux-system-roles/logging/commit/03b00f557cc525e54eebb3af3bc6b3d7f210009e#diff-7eb2a312068fa9ec1be5e435acb86b2def3a99f30b6ff3092f42964283bd9fa8) and this one. There were almost no difference... But probably, we need some way to avoid the risk to bring in an unexpected bug... Let me think about it...

Right.  The role should do nothing if all we want to do is set some vars.